pkg/logger/ecs_error_log.go

```diff
@@ -0,0 +1,175 @@
+package logger
+
+import (
+	"fmt"
+	"go.uber.org/zap/zapcore"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+)
+
+var ErrorFilePath = os.Getenv("STATUS_MESSAGE_FILE_PATH")
+
+var errorLogMaxAge = os.Getenv("STATUS_MESSAGE_TTL")
+var errorLogMaxAgeInHours, _ = strconv.Atoi(errorLogMaxAge)
+
+var size = os.Getenv("STATUS_MESSAGE_MAX_BYTE_LENGTH")
+var errorFileSize, _ = strconv.Atoi(size)
+
+var queue []zapcore.Entry
+var errorLogChannel = make(chan zapcore.Entry, 10)
+var ticker = time.NewTicker(10 * time.Minute)
+var errorFile *os.File
```

Comment:
Why is this a global variable? I think this is again an argument for my idea of putting everything in a new structure. 

Also, it looks like you're keeping this file open between calls. I am not sure if that is ideal. @mxiamxia @pingleig Is that bad? Keeping the file open between write calls? See the code in `func WriteError`; it keeps the file pointer open between calls.

Reply:
yeah, if we want to build all the things in an ECSErrorLogger type, this will be more clear, those variables will belongs to the type, and so does that error log file, just like everything set in the lumberjacklogger